src/main/kotlin/com/toasttab/pulseman/state/Tab.kt

```diff
@@ -1,159 +1,23 @@
-            selectedAuthClass = authSelector.selectedAuthClass.selected?.cls?.name,
-            authJsonParameters = authSelector.authJsonParameters(),
-            propertyMap = propertySettings.propertyMap()
-        )
-
-        if (save) {
-            lastSavedTabValues = tabValues
-        }
-
-        return tabValues
-    }
-
-    init {
-        initialSettings?.tabName?.let {
-            tabName.value = it
-        }
-    }
-
-    val isActive: Boolean
-        get() = selection.selected === this
-
-    fun activate() {
-        selection.selected = this
-    }
-
-    val pulsarSettings =
-        PulsarSettings(
-            appState = appState,
-            setUserFeedback = userFeedback::setUserFeedback,
-            initialSettings = initialSettings,
-            protobufSelector = protobufSelector,
-            authSelector = authSelector,
-            propertySettings = propertySettings,
-            onChange = ::onChange
-        )
-
-    val selectedView = mutableStateOf(SelectedView.SEND)
-
-    val sendMessage = SendMessage(
-        setUserFeedback = userFeedback::setUserFeedback,
-        selectedClass = protobufSelector.selectedSendClass,
-        pulsarSettings = pulsarSettings,
-        initialSettings = initialSettings,
-        onChange = { onChange() }
-    )
-    val receiveMessage = ReceiveMessage(
-        setUserFeedback = userFeedback::setUserFeedback,
-        pulsarSettings = pulsarSettings,
-        selectedReceiveClasses = protobufSelector.selectedReceiveClasses
-    )
-
-    // Tab icon settings
-    val focused = mutableStateOf(false)
-    val image = mutableStateOf(Icons.Default.Circle)
-    val drawBackground = mutableStateOf(false)
-
-    fun onEnterIconUnsavedChanges() {
-        drawBackground.value = true
-        image.value = Icons.Default.Close
-    }
-
-    fun onExitIconUnsavedChanges() {
-        drawBackground.value = false
-        image.value = Icons.Default.Circle
-    }
-
-    fun onEnterIcon() {
-        drawBackground.value = true
-    }
-
-    fun onExitIcon() {
-        drawBackground.value = false
-    }
-
-    fun onFocusedUpdate(newValue: Boolean) {
-        focused.value = newValue
-    }
-
-    fun onTabNameChange(newValue: String) {
-        tabName.value = newValue
-        onChange()
-    }
-
-    private fun onChange() {
-        unsavedChanges.value = lastSavedTabValues != tabValues()
-    }
-}
+data class Tab(
+    val tabName: String,
+    val close: () -> Unit,
+    val unsavedChanges: Boolean,
+    val initialSettings: TabValues?,
+    val isActive: Boolean,
+    val onFocusedUpdate: (Boolean) -> Unit,
+    val activate: () -> Unit,
+    val image: ImageVector,
+    val onEnterIconUnsavedChanges: () -> Unit,
+    val onExitIconUnsavedChanges: () -> Unit,
+    val onEnterIcon: () -> Unit,
+    val onExitIcon: () -> Unit,
+    val drawBackground: Boolean,
+    val focused: Boolean,
+    val ui: @Composable () -> Unit
+)
```

Comment:
A lot of the code in this class has been moved into TabState.
I wanted to pass around a stateless version of the current Tab that couldn't be modified.